src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -2961,10 +2970,10 @@ void addProperties(ImmutableArray<ParameterSymbol> recordParameters)
                 }
             }
 
-            void addObjectEquals()
+            void addObjectEquals(MethodSymbol thisEquals)
             {
-                var objEquals = new SynthesizedRecordObjEquals(this);
-                if (!memberSignatures.Contains(objEquals))
+                var objEquals = new SynthesizedRecordObjEquals(this, thisEquals);
+                if (!memberSignatures.ContainsKey(objEquals))
```

Comment:
Why bother with `ContainsKey`? Just `Add`, `Add` returns false if it's a duplicate.

Reply:
Same with all the others here, adjusting logic as appropriate. I don't think any of the `ContainsKey` methods are required.

---
In reply to: [413327921](https://github.com/dotnet/roslyn/pull/43133#discussion_r413327921) [](ancestors = 413327921)

Reply:
Those are two different collections. MemberSignatures is a dictionary, but `members` is a simple ArrayBuilder that we're appending things to.

Reply:
Ah, I misread that.